client/db/src/lib.rs

```diff
@@ -317,6 +317,18 @@ impl DatabaseSettingsSrc {
 	}
 }
 
+impl std::fmt::Display for DatabaseSettingsSrc {
+	fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+		let name = match self {
+			DatabaseSettingsSrc::RocksDb { .. } => "RocksDb",
+			DatabaseSettingsSrc::ParityDb { .. } => "ParityDb",
+			DatabaseSettingsSrc::SubDb { .. } => "SubDb",
+			DatabaseSettingsSrc::Custom(_) => "Custom",
+		};
+		write!(f, "{}", name)
+	}
+}
```

Comment:
You could make this smarter and display specific useful information per variant. By that I mean you can display the path like this `RocksDb (path: /x/y/z)` and that would also allow other variants to display different kind of useful information.

It's just an idea.

Reply:
That sounds more like job of `Debug`. `Display` should keep things to a minimum. It is hard to guess what format would be required in advance, so the actual formatting better be left to the calling code.

Reply:
I guess you're right